pkg/controllers/garbagecollector/garbagecollector.go

```diff
@@ -232,7 +233,12 @@ func (gc *gccontroller) processJob(key string) error {
 		Preconditions:     &metav1.Preconditions{UID: &fresh.UID},
 	}
 	klog.V(4).Infof("Cleaning up Job %s/%s", namespace, name)
-	return gc.vcClient.BatchV1alpha1().Jobs(fresh.Namespace).Delete(context.TODO(), fresh.Name, options)
+	err = gc.vcClient.BatchV1alpha1().Jobs(fresh.Namespace).Delete(context.TODO(), fresh.Name, options)
+	if err != nil && apierrors.IsNotFound(err) {
```

Comment:
How about
```go
if apierrors.IsNotFound(err) {
    // if the job had deleted, it will not be added to queue
	return nil
}
return err
```

Reply:
GOOD, I modified it according to your way.